src/librustc/hir/lowering.rs

```diff
@@ -4096,8 +4096,11 @@ impl<'a> LoweringContext<'a> {
                 };
 
                 // `mut iter`
-                let iter_pat =
-                    self.pat_ident_binding_mode(head_sp, iter, hir::BindingAnnotation::Mutable);
+                let iter_pat = self.pat_ident_binding_mode(
+                    desugared_span,
```

Comment:
Must be because this is now using `desugared_span` and not `head.span` — we could perhaps "normalize" the spans we use to suppress duplicate trait errors by skipping over "compiler desugarings" into the underlying span.

Reply:
@nikomatsakis For my understanding could you clarify what does normalizing spans mean ?

Reply:
@varkor Thanks a lot for the explanation. Makes sense.